wordpress-mobile/WordPress-Android
# Issue/13329 backup download error retry

Parent #13329 

This PR updates `GetBackupDownloadStatusUseCase` with the following:
(1) Reorders execution so that the Fetch from the server is first, then a get on the local db
(2) Enable 3 retries when error is received from server fetch
(3) Lowers the delay to 5 seconds (from 10)
(4) Emits an `Empty` state when an null status model is returned (Note: The FluxC part has not been implemented yet)

This PR also includes a new test class: `GetBackupDownloadStatusUseCaseTest`

**To test:**
-Run the app and make sure the backup/download still functions as expected

PR submission checklist:
